In [1]:
import pandas as pd

In [2]:
sales = pd.read_csv("sales.csv")

In [3]:
%%sql
SELECT
    client_type,
    SUM(total) AS total_sales
FROM
    sales
GROUP BY
    client_type;

client_type  total_sales
0   Wholesale    159642.33
1      Retail    129470.67

In [4]:
%%sql
SELECT
    client_type,
    AVG(total) AS avg_order_value
FROM
    sales
GROUP BY
    client_type;

client_type  avg_order_value
0   Wholesale       709.521467
1      Retail       167.058929

In [7]:
%%sql
SELECT
    client_type,
    product_line,
    SUM(quantity) AS total_quantity
FROM
    sales
GROUP BY
    client_type, product_line
ORDER BY
    total_quantity DESC
LIMIT 10;

client_type           product_line  total_quantity
0   Wholesale        Breaking system          1268.0
1   Wholesale  Suspension & traction          1128.0
2      Retail  Suspension & traction          1017.0
3   Wholesale           Frame & body           932.0
4      Retail        Breaking system           862.0
5      Retail      Electrical system           862.0
6   Wholesale      Electrical system           836.0
7      Retail           Frame & body           687.0
8   Wholesale          Miscellaneous           668.0
9      Retail          Miscellaneous           508.0

In [9]:
%%sql
SELECT
    client_type,
    payment,
    COUNT(*) AS count
FROM
    sales
GROUP BY
    client_type, payment
ORDER BY
    count DESC;

client_type      payment  count
0      Retail  Credit card    659
1   Wholesale     Transfer    225
2      Retail         Cash    116

In [10]:
%%sql
SELECT
    client_type,
    COUNT(DISTINCT order_number) AS order_count
FROM
    sales
GROUP BY
    client_type;

client_type  order_count
0      Retail          775
1   Wholesale          225

In [11]:
%%sql
SELECT
    client_type,
    SUM(total) AS total_revenue
FROM
    sales
GROUP BY
    client_type;

client_type  total_revenue
0   Wholesale      159642.33
1      Retail      129470.67

# Client Type Analysis

This report provides an analysis of sales performance and purchasing behavior between retail and wholesale clients based on the provided dataset.

## Overview

The client type analysis aims to understand the differences in sales volume, purchasing behavior, and revenue generation between retail and wholesale clients.

## Findings

### Sales Volume and Revenue

- **Retail Clients**:
  - Total sales volume: $129,470.67
  - Total revenue generated: $129,470.67

- **Wholesale Clients**:
  - Total sales volume: $159,642.33
  - Total revenue generated: $159,642.33

### Observations

1. **Sales Volume**:
   - Wholesale clients have a higher sales volume compared to retail clients, indicating larger quantities purchased by wholesale clients, likely due to bulk orders.
   - Retail clients, while having lower sales volume, still contribute significantly to overall sales.

2. **Purchasing Behavior**:
   - Retail clients demonstrate a higher frequency of purchases, making more frequent but possibly smaller orders compared to wholesale clients.
   - Wholesale clients make fewer but likely larger orders, reflecting the nature of wholesale transactions involving bulk purchases.

3. **Revenue Generation**:
   - Wholesale clients generate higher revenue compared to retail clients, indicating that wholesale transactions contribute more to the overall revenue.

## Conclusion

The client type analysis highlights the differences in sales volume, purchasing behavior, and revenue generation between retail and wholesale clients. Understanding these differences is essential for tailoring marketing strategies, inventory management, and pricing strategies to optimize profitability and overall business performance.